docs/src/components/icons-list.astro

```diff
@@ -2,10 +2,18 @@
 import { Icon } from '@astrojs/starlight/components';
 import { Icons } from '../../../packages/starlight/components/Icons';
 
+interface Props {
+	labels?: {
+		copied?: string;
+	};
+}
+
+const { copied = 'Copied!' } = Astro.props.labels ?? {};
+
 const icons = Object.keys(Icons) as (keyof typeof Icons)[];
 ---
 
-<div class="icons-grid">
+<div class="icons-grid" data-label-copied={copied}>
```

Comment:
The "copied" label is added only once to the DOM and not for every icons.

Reply:
Perfect. We could even delegate the click handling here too I guess to avoid assigning a listener for every button? Doesn’t have to be in this PR though!